pkg/cloud/openstack/openstack.go

```diff
@@ -80,3 +84,73 @@ func NewProviderAssets(config config.OperatorConfig) (common.CloudProviderAssets
 
 	return assets, nil
 }
+
+// CloudConfigTransformer implements the cloudConfigTransformer. It uses the source ConfigMap and
+// to create a new config that adheres to the configuration for CCM
+// It returns an error if the platform is not OpenStackPlatformType.
+func CloudConfigTransformer(source *corev1.ConfigMap, infra *configv1.Infrastructure) (*corev1.ConfigMap, error) {
+	if !(infra.Status.PlatformStatus != nil &&
+		infra.Status.PlatformStatus.Type == configv1.OpenStackPlatformType) {
+		return nil, fmt.Errorf("invalid platform, expected to be OpenStack")
+	}
+
+	output := source.DeepCopy()
+	infraConfigKey := infra.Spec.CloudConfig.Key
+	if val, ok := output.Data[infraConfigKey]; ok {
+		output.Data[common.DefaultConfigKey] = val
+		delete(output.Data, infraConfigKey)
+	} else {
+		return nil, fmt.Errorf(
+			"key %s specified in infra resource was not found in source configmap %s",
+			infraConfigKey, client.ObjectKeyFromObject(source),
+		)
+	}
+
+	// TODO: Do we need to care about stuff in output.BinaryData?
+	var inCfgRaw []byte
+	if v, ok := output.Data[common.DefaultConfigKey]; ok {
+		inCfgRaw = []byte(v)
+	}
+
+	cfg, err := ini.Load(inCfgRaw)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read the cloud.conf: %w", err)
+	}
+
+	// TODO: Use 'HasSection' when we bump to gopkg.in/ini.v1 v1.67 or later
```

Comment:
Why can't we bump to that now?

Reply:
We don't need it anyway. If there was a HasSection we'd write:

```
if HasSection {
  GetSection
}
```

Without it we write what we've got here:

```
GetSection
if section == nil {...}
```

It would be redundant.

Reply:
v1.67 doesn't exist yet. I can remove this `TODO`. It's not important.